doc/reference/configuration/configuration_reference.rst

```diff
@@ -2174,3 +2174,340 @@ The ``security`` section defines configuration parameters related to various sec
+    :doc:`box.snapshot() </reference/reference_lua/box_snapshot>` every ``snapshot.by.interval``
+    seconds, creating a new snapshot file each time.
+    If the option is set to zero, the checkpoint daemon is disabled.
+
+    Example:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/persistence_snapshot/config.yaml
+        :language: yaml
+        :start-at: by:
+        :end-at: interval: 60
+        :dedent:
+
+    In the example, the checkpoint daemon creates a new database snapshot once per minute, if there is activity.
+
+    |
+    | Type: number
+    | Default: 3600
+    | Environment variable: TT_SNAPSHOT_BY_INTERVAL
+
+..  _configuration_reference_snapshot_by_wal_size:
+
+..  confval:: snapshot.by.wal_size
+
+    The threshold for the total size in bytes of all WAL files created since the last snapshot taken.
+    Once the configured threshold is exceeded, the WAL thread notifies the
+    :ref:`checkpoint daemon <configuration_persistence_checkpoint_daemon>` that it must make a new snapshot and delete old WAL files.
+
+    |
+    | Type: integer
+    | Default: 10^18
+    | Environment variable: TT_SNAPSHOT_BY_WAL_SIZE
+
+..  _configuration_reference_wal:
+
+wal
+---
+
+The ``wal`` section defines configuration parameters related to :ref:`write-ahead log <internals-wal>`.
+To learn more about the WAL configuration, check the :ref:`Persistence <configuration_persistence_wal>` page.
+
+..  NOTE::
+
+    ``wal`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+-   :ref:`wal.cleanup_delay <configuration_reference_wal_cleanup_delay>`
+-   :ref:`wal.dir <configuration_reference_wal_dir>`
+-   :ref:`wal.dir_rescan_delay <configuration_reference_wal_dir_rescan_delay>`
+-   :ref:`wal.max_size <configuration_reference_wal_max_size>`
+-   :ref:`wal.mode <configuration_reference_wal_mode>`
+-   :ref:`wal.queue_max_size <configuration_reference_wal_queue_max_size>`
+-   :ref:`wal.ext.* <configuration_reference_wal_ext>`
+
+    -   :ref:`wal.ext.new <configuration_reference_wal_ext_new>`
+    -   :ref:`wal.ext.old <configuration_reference_wal_ext_old>`
+    -   :ref:`wal.ext.spaces <configuration_reference_wal_ext_spaces>`
+
+..  _configuration_reference_wal_cleanup_delay:
+
+..  confval:: wal.cleanup_delay
+
+    The delay in seconds used to prevent the :ref:`checkpoint daemon <configuration_persistence_checkpoint_daemon>`
+    from immediately removing :ref:`write-ahead log <internals-wal>` files after a node restart.
+    This delay eliminates possible erroneous situations when the master deletes WALs
+    needed by :ref:`replicas <replication-roles>` after restart.
+    As a consequence, replicas sync with the master faster after its restart and
+    don't need to download all the data again.
+
+    Once all the nodes in the replica set are up and running,
+    automatic cleanup is started again even if ``wal.cleanup_delay`` has not expired.
```

Comment:
I guess that the idea is that a usual garbage collection schedule is in effect after all the replicas become up. Not that a clean up is performed immediatelly, when they're become up. (It would be good to re-check it with @sergepetrenko.)

Reply:
Rewrote the sentence to make it more clear